src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -682,6 +684,171 @@
 )
 
 
+class ObjectReferenceSegment(ansi.ObjectReferenceSegment):
+    """A reference to an object.
+
+    Overriding ObjectReferenceSegment to support Snowflake's IDENTIFIER pseudo-function.
+    """
+
+    type = "object_reference"
+    match_grammar: Matchable = OneOf(
+        # Snowflake's IDENTIFIER pseudo-function
+        # https://docs.snowflake.com/en/sql-reference/identifier-literal.html
+        Delimited(
+            Sequence(
+                "IDENTIFIER",
+                Bracketed(
+                    OneOf(
+                        Ref("SingleQuotedIdentifierSegment"),
+                        Ref("ReferencedVariableNameSegment"),
+                    ),
+                ),
+            ),
+        ),
```

Comment:
I don't think you need the `Delimited` here. That's only used in the next section to allow "dot" delimiters (e.g. `schema.table.column`). As you are not overriding the delimiter here from the default comma, it's not needed.

```suggestion
        Sequence(
            "IDENTIFIER",
            Bracketed(
                OneOf(
                    Ref("SingleQuotedIdentifierSegment"),
                    Ref("ReferencedVariableNameSegment"),
                ),
            ),
        ),
```

Though I wonder if `schema.table.IDENTIFIER('column')` is valid? I presume not?

Reply:
From the docs, it doesn't seem likely. But they aren't explicit. It might be a valid edge case, but I'm guessing that it's not.

As it happens... this variation currently parses fine:
`create or replace table my_db.my_schema.identifier('my_table') (c1 number);`
I have not added a test case, since I don't know if it's truly valid. But it's more likely than your column example.